server/handlers/handlers.go

```diff
@@ -553,3 +554,33 @@ func (ah *ActivityHandler) isRemoteUser(userID string) bool {
 func IsDirectMessage(chatID string) bool {
 	return chatID != ""
 }
+
+func (ah *ActivityHandler) SyncChannelSince(teamID, channelID string, syncSince time.Time) error {
+	return ah.plugin.GetClientForApp().OnChannelMessagesSince(teamID, channelID, syncSince, func(message *clientmodels.Message) {
+		isCreation := false
+		if message.CreateAt == message.LastUpdateAt {
+			isCreation = true
```

Comment:
I know we talked about idempotency below, but it feels weird to make an `isCreation` decision here. If I sync back 24 hours, and I find a post that was edited after creation, I unilaterally call `handleUpdatedActivity`.

But if that post was genuinely new to Mattermost, doesn't `handleUpdatedActivity` discard it because it can't find the post info?

https://github.com/mattermost/mattermost-plugin-msteams-sync/blob/f586fb8c88f80c3cce6c2fb546f9cebca05aa954/server/handlers/handlers.go#L358-L361

Reply:
I don't think your claim is right here.

If you have a post that has been edited after creation, it falls in the "else", that checks if that message is already in mattermost, so it would go to `handleCreateActivity` if that doesn't exists. If the post is a creation it is going to go always to the `handleCreateActivity`.

What I think is wrong, is the fact that I'm not passing the ChatID/ChannelID in the `GetPostInfoByMSTeamsID`, so I think I have to fix that.

Reply:
Ah, you're right, I didn't follow that logic through -- would it make sense to just unconditionally call `GetPostInfoByMSTeamsID` to decide `isCreation` instead of doing `message.CreateAt == message.LastUpdateAt` at all? While it means unconditionally making an API call, I wonder if it would help prevent a scenario where we somehow get a replayed message from MS Teams and try to create it instead of just "no-op" updating it.

Reply:
I guess so, I think we can add back the "optimization" later if it is needed.

Reply:
Done